app/backend/app.py

```diff
@@ -276,19 +355,24 @@ async def setup_clients():
         index_name=AZURE_SEARCH_INDEX,
         credential=search_credential,
     )
-    search_index_client = SearchIndexClient(
-        endpoint=f"https://{AZURE_SEARCH_SERVICE}.search.windows.net",
-        credential=search_credential,
-    )
 
     blob_client = BlobServiceClient(
         account_url=f"https://{AZURE_STORAGE_ACCOUNT}.blob.core.windows.net", credential=azure_credential
     )
     blob_container_client = blob_client.get_container_client(AZURE_STORAGE_CONTAINER)
+    await blob_client.close()
 
     # Set up authentication helper
+    search_index = None
+    if AZURE_USE_AUTHENTICATION:
+        search_index_client = SearchIndexClient(
+            endpoint=f"https://{AZURE_SEARCH_SERVICE}.search.windows.net",
+            credential=search_credential,
+        )
+        search_index = await search_index_client.get_index(AZURE_SEARCH_INDEX)
+        await search_index_client.close()
```

Comment:
This was never closed before :(

Reply:
theoretically we can use async with as well here